### PR TITLE
2885 Load lookup tables locally for testing

### PR DESCRIPTION
**Pull request by carlosfelix2**
This PR adds functionality to the CLI to load a series of tables from a folder and activate the tables to make it easy to load tables in our local environments.  It also adds the automated loading of the tables when starting the local container.  Note that this new load command is not allowed in production, just so there are no mistakes.

Test Steps:
Currently there are no tables being stored in the database, so we will use some of the existing ones for this test.
1. Copy two or more tables from metadata/tables to src/test/resources/metadata/tables.  Rename the extensions from .tables to .csv.
2. Run `gradle package` and run your functions.
3. If you run the functions in the docker container then verify the tables were loaded automatically by running `../prime lookuptables list` and verifying the tables you added are listed.
4. Run the CLI to load add the tables: `./prime lookuptables loadall`
5. Verify the tables were loaded by getting the list of tables with `./prime lookuptables list`.  If your container loaded the tables already then verify a new version was added.

## Changes
- CLI can load a full folder of tables
- Tables are loaded during the start of the local container
- Added gradle task to load the tables
- Changed gradle task reloadDb to resetDb to try to make it obvious what it does
- Changed the table creation to batched which made large table uploads.  NCIS table (12200+ rows) went from 5 mins to 25 secs

### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Fixes
- #issue - List GitHub issues this PR fixes

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: prime-router/start_func.sh [hunk 1]
```diff
@@ -23,4 +23,10 @@ fi
 
 cd $function_folder
 
+# Use the prime CLI to load configuration in the background
+# The fatjar is two levels up in the libs folder
+top_dir=$function_folder/../..
+fatjar=$top_dir/libs/prime-router-0.1-SNAPSHOT-all.jar
+java -jar $fatjar lookuptables loadall -d $top_dir/resources/test/metadata/tables -r 10 | awk '{print "[LOAD TABLES] " $0}' &
```
Comment (carlosfelix2) [inline prime-router/start_func.sh:30]: Some notes here:
1. I use awk to add text, so it is easy to see the outout.
2. I am aiming to the same thing here for the settings table.  I would like to remove the use of the settings container to save memory.  Thoughts?

**Review by MauriceReeves-usds — APPROVED** [review]

File: prime-router/src/main/kotlin/cli/LookupTableCommands.kt [hunk 2]
```diff
@@ -724,4 +790,68 @@ class LookupTableActivateCommand : GenericLookupTableCommand(
...
+    private val connRetries by option("-r", "--retries", help = "Connection retries when checking the endpoint")
+        .int().default(10)
+
+    /**
+     * The reference to the table creator command.
+     */
+    private val tableCreator = LookupTableCreateCommand()
+
+    override fun run() {
+        if (environment == Environment.PROD) error("This command is not allowed for the production environment.")
```
Comment (carlosfelix2) [inline prime-router/src/main/kotlin/cli/LookupTableCommands.kt:829]: Change to only allow in local.

File: prime-router/start_func.sh (hunk 1 above)
Comment (carlosfelix2) [reply prime-router/start_func.sh:30]: Check on loading the tables only when they change.

**Review by jimduff-usds — APPROVED** [review]
Approved once its changed to only update when localhost file tables have changed.

File: prime-router/start_func.sh (hunk 1 above)
Comment (clediggins-usds) [reply prime-router/start_func.sh:30]: Agreed - it would be great if this did not load on every docker-up unless changes are present.

**Review by clediggins-usds — CHANGES_REQUESTED** [review]
Comments added after conversation/PR review.

**Review by clediggins-usds — APPROVED** [review]
Looks good. Thanks

**Pull request merged**